xdsl/traits.py

```diff
@@ -110,3 +112,27 @@ def verify(self, op: Operation) -> None:
                     # too; in which case it will check itself.
                     if not child_op.has_trait(IsolatedFromAbove):
                         regions += child_op.regions
+
+
+class SymbolOpInterface(OpTrait):
+    @staticmethod
+    def get_sym_attr_name(op: Operation) -> StringAttr:
+        """
+        Returns the symbol of the operation
+        """
+        return cast(StringAttr, op.attributes["sym_name"])
```

Comment:
I'm not a huge fan of cast, I would prefer an `isa` guard instead, as cast won't fail at runtime.

`cast` is basically `type: ignore`

Reply:
That was my intended behaviour, the verification below already makes sure that it's a StrinAttr, I don't think we should be checking things in two places.

Reply:
I still think an additional assert is preferable to code that typechecks successfully but still fails (as there are plenty of cases when IR is unverified or in a broken state and still passed around).

I'd rather have the code blow up here, than somewhere down the line.

Reply:
I think the interpret should have a clear assumption that the IR is verified, and not check everything again, too. That's the point of even having verifiers and verification infrastructure, I believe. All passes also work on the contract that they take verified IR.

Reply:
Off course this should be reflected in the `xdsl-run` tool, which should make sure to verify the Module before interpreting anything.

Reply:
Oh wow totally missed that, my bad!